addons/binding/org.openhab.binding.innogysmarthome/src/main/java/org/openhab/binding/innogysmarthome/internal/manager/DeviceStructureManager.java

```diff
@@ -0,0 +1,348 @@
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.innogysmarthome.internal.manager;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.openhab.binding.innogysmarthome.InnogyBindingConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import in.ollie.innogysmarthome.InnogyClient;
+import in.ollie.innogysmarthome.entity.Location;
+import in.ollie.innogysmarthome.entity.Message;
+import in.ollie.innogysmarthome.entity.Property;
+import in.ollie.innogysmarthome.entity.capability.Capability;
+import in.ollie.innogysmarthome.entity.device.Device;
+import in.ollie.innogysmarthome.entity.link.CapabilityLink;
+import in.ollie.innogysmarthome.exception.ApiException;
+
+/**
+ * Manages the structure of the {@link Device}s and the calls to the {@link InnogyClient} to load the {@link Device}
+ * data from the innogy SmartHome web service.
+ *
+ * @author Oliver Kuhl - Initial contribution
+ *
+ */
+public class DeviceStructureManager {
+
+    private Logger logger = LoggerFactory.getLogger(DeviceStructureManager.class);
+
+    private InnogyClient client;
+    private Map<String, Device> deviceMap;
+    private Map<String, Device> capabilityToDeviceMap;
+    private Map<String, Location> locationMap;
+    private String bridgeDeviceId;
+
+    /**
+     * Constructs the {@link DeviceStructureManager}.
+     *
+     * @param client the {@link InnogyClient}
+     */
+    public DeviceStructureManager(InnogyClient client) {
+        this.client = client;
+    }
+
+    /**
+     * Starts the {@link DeviceStructureManager} by building the device structure.
+     *
+     * @throws IOException
+     * @throws ApiException
+     */
+    public synchronized void start() throws IOException, ApiException {
+        logger.debug("Starting device structure manager.");
+
+        refreshDevices();
+        logger.info("Devices loaded. Device structure manager ready.");
+        writeExampleConfigToLog();
+    }
+
+    /**
+     * Returns the {@link #deviceMap}, a map with the device id and the device.
+     *
+     * @return
+     */
+    public Map<String, Device> getDeviceMap() {
+        if (deviceMap == null) {
+            deviceMap = Collections.synchronizedMap(new HashMap<String, Device>());
+        }
+        return deviceMap;
+
+    }
+
+    /**
+     * Loads all device data from the bridge and stores the {@link Device}s and their states in the
+     * {@link DeviceStructureManager}.
+     *
+     * @throws IOException
+     * @throws ApiException
+     */
+    public void refreshDevices() throws IOException, ApiException {
```

Comment:
why? it never gets called from outside, it does not return anything, it modifies internal state and it probably needs some time to run since it calls `client.getFullDevices()`.

Reply:
Done.